pkg/providers/aws/credentials.go

```diff
@@ -136,14 +137,24 @@ type Credentials struct {
 	PolicyName      string
 	AccessKeyID     string
 	SecretAccessKey string
+	RoleArn         string
+	TokenFilePath   string
 }
 
 //go:generate moq -out credentials_moq.go . CredentialManager
 type CredentialManager interface {
 	ReconcileProviderCredentials(ctx context.Context, ns string) (*Credentials, error)
-	ReconcileSESCredentials(ctx context.Context, name, ns string) (*Credentials, error)
-	ReoncileBucketOwnerCredentials(ctx context.Context, name, ns, bucket string) (*Credentials, *v1.CredentialsRequest, error)
-	ReconcileCredentials(ctx context.Context, name string, ns string, entries []v1.StatementEntry) (*v1.CredentialsRequest, *Credentials, error)
+	ReconcileBucketOwnerCredentials(ctx context.Context, name, ns, bucket string) (*Credentials, error)
+}
+
+func NewCredentialManager(client client.Client) CredentialManager {
```

Comment:
This appears to be used mostly in the aws providers but I see two entries where it's used in controllers. postgressnapshot and redissnapshot controllers. Should the controllers be provider independent?

Reply:
Good catch.
It seems like `CredentialManager` field in those two controllers is unused, so I'll just remove it.